python/pyspark/tests/test_rddsampler.py

```diff
@@ -0,0 +1,67 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from pyspark.testing.utils import ReusedPySparkTestCase
+from pyspark.rddsampler import RDDSampler, RDDStratifiedSampler
+
+
+class RDDSamplerTests(ReusedPySparkTestCase):
+    def test_rdd_sampler_func(self):
+        # SPARK-38879:Test case to improve test coverage for RDDSampler
+        # RDDSampler.func
+        rdd = self.sc.parallelize(range(20), 2)
+        sample_count = rdd.mapPartitionsWithIndex(RDDSampler(False, 0.4, 10).func).count()
+        self.assertGreater(sample_count, 4)
+        self.assertLess(sample_count, 10)
+        sample_data = rdd.mapPartitionsWithIndex(RDDSampler(True, 0.4, 10).func).collect()
+        sample_data.sort()
+        is_repetition_contain = False
+        # check if at least one element is repeated.
+        for index in range(1, len(sample_data)):
```

Comment:
I think you can reread this part like:

```suggestion
self.assertTrue(any(
    sample_data[i] == sample_data[i - 1] for i in range(1, len(sample_data))))
```